packages/pymedphys_dicom/src/pymedphys_dicom/dicom/anonymise.py

```diff
@@ -579,9 +528,16 @@ def unknown_tags_in_dicom_dataset(ds):
 
     non_private_tags_in_dataset = np.array(non_private_tags_in_dicom_dataset(ds))
 
-    are_non_private_tags_in_dict_baseline = [
-        tag in BaselineDicomDictionary.keys() for tag in non_private_tags_in_dataset
-    ]
+    are_non_private_tags_in_dict_baseline = []
+    for tag in non_private_tags_in_dataset:
+        try:
+            get_baseline_dict_entry(tag)
+            are_non_private_tags_in_dict_baseline.append(True)
+        except KeyError as ek:
+            if "not found in DICOM dictionary" in str(ek):
+                are_non_private_tags_in_dict_baseline.append(False)
+            else:
+                raise
```

Comment:
For this pattern I believe the appropriate way to go about it is to create our own exception and then Rai's an except that sees that exception. It would certainly be odd down the road changing the wording of an error message and have it break logic. It's an odd thing to depend on.

Instead let's create an exception called `NotInBaseline` or something like that, raise that in that function on the line where that key error happens, and then catch our custom exception instead.

Reply:
Yeah, you're right. Done. :) See https://github.com/pymedphys/pymedphys/pull/437/commits/b10d1ab40aa958ccddff3c732d30a1fce55f2983